packages/code-du-travail-data/indexing/addGlossary.js

```diff
@@ -0,0 +1,56 @@
+import { v4 as uuidv4 } from "uuid";
+const glossary = require("@socialgouv/datafiller-data/data/glossary.json");
+
+const conventionMatchers = [
+  "convention collective",
+  "conventions collectives",
+  "accords de branches",
+  "accord de branche",
+  "disposition conventionnelle",
+  "dispositions conventionnelles",
+];
+
+export function addGlossary(htmlContent) {
+  const internalRefMap = new Map();
+  const formatedContent = glossary.reduce((updatedContent, match) => {
+    const { abbrs, title, definition, variants } = match;
+    if (!updatedContent) return updatedContent;
+    // we cannot use \b word boundary since \w does not match diacritics
+    // So we do a kind of \b equivalent.
+    // the main différence is that matched pattern can include a whitespace as first char
+    const frDiacritics = "àâäçéèêëïîôöùûüÿœæÀÂÄÇÉÈÊËÎÏÔÖÙÛÜŸŒÆ";
+    const wordBoundaryStart = `(?:^|[^_/\\w${frDiacritics}-])`;
+    const wordBoundaryEnd = `(?![\\w${frDiacritics}])`;
+    const patterns = [...new Set([title, ...variants])].map(
+      (term) =>
+        new RegExp(`${wordBoundaryStart}(${term})${wordBoundaryEnd}`, "gi")
+    );
+    if (abbrs) {
+      patterns.concat(new RegExp(`\\b(${abbrs})\\b`, "g"));
```

Comment:
tiens d'ailleurs attention, concat renvoie un nouveau tableau, c'est pas in place donc là on fait le concat dans le vent

Reply:
un .push devrait suffire